pyrk/th_component.py

```diff
@@ -35,27 +42,68 @@ def __init__(self, name=None,
         :type timer: Timer object
         :param heatgen: is this component a heat generator (fuel)
         :type heatgen: bool
+        :param power_tot: power generated in this component
+        :type power_tot: float
+        :param sph: is this component a spherical component, spherical equations
+        for heatgen, conduction are different, post-processing is different too
+        :type sph: bool
+        :param ri: inner radius of the sph/annular component, ri=0 for sphere
+        :type ri: float
+        :param ro: outer radius of the sph/annular component, ro=radius for sphere
+        :type ro: float
         """
         self.name = name
-        self.vol = vol.to('meter**3')
-        validation.validate_ge("vol", vol, 0*units.meter**3)
+        self.vol = vol
+        self.mat = mat
         self.k = mat.k
         self.cp = mat.cp
         self.dm = mat.dm
-        self.T0 = T0.to('kelvin')
-        validation.validate_num("T", T0)
+        self.timer = timer
         self.T = units.Quantity(np.zeros(shape=(timer.timesteps(),),
                                          dtype=float), 'kelvin')
         self.T[0] = T0
+        self.T0 = T0
         self.alpha_temp = alpha_temp.to('delta_k/kelvin')
-        self.timer = timer
         self.heatgen = heatgen
         self.power_tot = power_tot
         self.cond = {}
         self.conv = {}
+        self.adv = {}
         self.mass = {}
         self.cust = {}
         self.prev_t_idx = 0
+        self.convBC = {}
+        self.sph = sph
+        self.ri = ri
+        self.ro = ro
+
+    def mesh(self, size):
+        '''cut a THComponent into a list of smaller components
+        uniform meshing method, only implemented for spherical components
+        :param size: size of uniform mesh element
+        :type size: float.
+        :return: list of components
+        '''
+        assert self.sph, '''mesh function only implemented for spherical component'''
```

Comment:
It might just be the leftover old image files. We probably should clear the
image folder before every plotting. I got tricked several times already.

On Fri, Sep 4, 2015 at 3:55 PM, Katy Huff <notifications@github.com> wrote:

> In pyrk/th_component.py
> <https://github.com/pyrk/pyrk/pull/10#discussion_r38800952>:
>
> >          self.mass = {}
> >          self.cust = {}
> >          self.prev_t_idx = 0
> > +        self.convBC = {}
> > +        self.sph = sph
> > +        self.ri = ri
> > +        self.ro = ro
> > +
> > +    def mesh(self, size):
> > +        '''cut a THComponent into a list of smaller components
> > +        uniform meshing method, only implemented for spherical components
> > +        :param size: size of uniform mesh element
> > +        :type size: float.
> > +        :return: list of components
> > +        '''
> > +        assert self.sph, '''mesh function only implemented for spherical component'''
>
> hm, yeah, I see what your'e saying. it really just shouldn't be possible.
> It was in the sfr examples. I'm re-running it now... just to be sure. I'll
> let you know.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyrk/pyrk/pull/10/files#r38800952>.
>



-- 
-----------------------------------------
Xin WANG | 王昕
PhD Student
Thermal Hydraulics Laboratory
Nuclear Engineering Department
University of California, Berkeley
(415)408-1691